src/sdk/formatting/formatBytes.test.ts

```diff
@@ -0,0 +1,80 @@
+		expect(formatBytes(10)).to.eql('10\xa0B');
+		expect(formatBytes(100)).to.eql('100\xa0B');
+		expect(formatBytes(100, { separator: ' '})).to.eql('100 B');
+		expect(formatBytes(100, { separator: ''})).to.eql('100B');
+	});
+
+	it('can format positive kb', () => {
+		expect(formatBytes(Math.pow(2, 10))).to.eql('1\xa0KB');
+		expect(formatBytes(Math.pow(2, 10) * 1.21)).to.eql('1.2\xa0KB');
+		expect(formatBytes(Math.pow(2, 10) * 9.8)).to.eql('9.8\xa0KB');
+	});
+
+	it('can format positive mb', () => {
+		expect(formatBytes(Math.pow(2, 20))).to.eql('1\xa0MB');
+		expect(formatBytes(Math.pow(2, 20) * 1.213)).to.eql('1.2\xa0MB');
+		expect(formatBytes(Math.pow(2, 20) * 9.81)).to.eql('9.8\xa0MB');
+	});
+
+	it('can format positive gb', () => {
+		expect(formatBytes(Math.pow(2, 30))).to.eql('1\xa0GB');
+		expect(formatBytes(Math.pow(2, 30) * 1.213)).to.eql('1.2\xa0GB');
+		expect(formatBytes(Math.pow(2, 30) * 9.81)).to.eql('9.8\xa0GB');
+	});
+
+	it('can format positive tb', () => {
+		expect(formatBytes(Math.pow(2, 40))).to.eql('1\xa0TB');
+		expect(formatBytes(Math.pow(2, 40) * 1.213)).to.eql('1.2\xa0TB');
+		expect(formatBytes(Math.pow(2, 40) * 9.81)).to.eql('9.8\xa0TB');
+	});
+
+	it('can format large positive tb', () => {
+		expect(formatBytes(Math.pow(2, 40) * 1024)).to.eql('1\xa0PB');
+		expect(formatBytes(Math.pow(2, 40) * 1.213 * 1024)).to.eql('1.2\xa0PB');
+		expect(formatBytes(Math.pow(2, 40) * 9.81 * 1024)).to.eql('9.8\xa0PB');
+	});
+
+	it('can format negative bytes', () => {
+		expect(formatBytes(-10)).to.eql('-10\xa0B');
+		expect(formatBytes(-100)).to.eql('-100\xa0B');
+	});
+
+	it('can format negative kb', () => {
+		expect(formatBytes(-Math.pow(2, 10))).to.eql('-1\xa0KB');
+		expect(formatBytes(-Math.pow(2, 10) * 1.21)).to.eql('-1.2\xa0KB');
+		expect(formatBytes(-Math.pow(2, 10) * 9.8)).to.eql('-9.8\xa0KB');
+	});
+
+	it('can format negative mb', () => {
+		expect(formatBytes(-Math.pow(2, 20))).to.eql('-1\xa0MB');
+		expect(formatBytes(-Math.pow(2, 20) * 1.21)).to.eql('-1.2\xa0MB');
+		expect(formatBytes(-Math.pow(2, 20) * 9.8)).to.eql('-9.8\xa0MB');
+	});
+
+	it('can format negative gb', () => {
+		expect(formatBytes(-Math.pow(2, 30))).to.eql('-1\xa0GB');
+		expect(formatBytes(-Math.pow(2, 30) * 1.21)).to.eql('-1.2\xa0GB');
+		expect(formatBytes(-Math.pow(2, 30) * 9.8)).to.eql('-9.8\xa0GB');
+	});
+
+	it('can format negative tb', () => {
+		expect(formatBytes(-Math.pow(2, 40))).to.eql('-1\xa0TB');
+		expect(formatBytes(-Math.pow(2, 40) * 1.21)).to.eql('-1.2\xa0TB');
+		expect(formatBytes(-Math.pow(2, 40) * 9.8)).to.eql('-9.8\xa0TB');
+	});
+
+	it('can format large negative tb', () => {
+		expect(formatBytes(-Math.pow(2, 40) * 1024)).to.eql('-1\xa0PB');
+		expect(formatBytes(-Math.pow(2, 40) * 1.213 * 1024)).to.eql('-1.2\xa0PB');
+		expect(formatBytes(-Math.pow(2, 40) * 9.81 * 1024)).to.eql('-9.8\xa0PB');
+	});
+
+	it('can format large yb', () => {
+		expect(formatBytes(Math.pow(2, 80) * 1024)).to.eql('1,024\xa0YB');
+	});
+});
```

Comment:
should there be a test for 0 and for a non number?

Reply:
Done